projects/libaom/av1_dec_fuzzer.cc

```diff
@@ -0,0 +1,91 @@
+// Fuzzing of AV1 decoder.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "aom/aom_decoder.h"
+#include "aom/aomdx.h"
+#include "aom_ports/mem_ops.h"
+#include "aom/common/ivfdec.h"
+
+// TODO(urvang): Create a 2nd fuzz target that defines DECODE_MODE_threaded.
+#define DECODE_MODE_serial
+
+static const char *const kIVFSignature = "DKIF";
+
+extern "C" void usage_exit(void) { exit(EXIT_FAILURE); }
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  FILE *const file = fmemopen((void *)data, size, "rb");
+  if (file == nullptr) {
+    fprintf(stderr, "Failed to get file stream.\n");
+    return 0;
+  }
+
+  char header[32];
+  if (fread(header, 1, 32, file) != 32) {
+    fprintf(stderr, "Failed to get file header.\n");
+    return 0;
+  }
+  if (memcmp(kIVFSignature, header, 4) != 0) {
+    fprintf(stderr, "Wrong IVF signature.\n");
+    // Do not terminate.
+  }
+  if (mem_get_le16(header + 4) != 0) {
+    fprintf(stderr, "Wrong IVF version.\n");
+    // Do not terminate.
+  }
+
+  const AvxInterface *decoder = get_aom_decoder_by_name("av1");
+  if (decoder == nullptr) {
+    fprintf(stderr, "Unknown input codec.\n");
```

Comment:
It's generally a bad idea to log for non-fatal errors in a fuzzer, because the logs will be spammed with these logs.

Can you change this so nothing is printed by your code when fuzzing please?

Also, please do the same elsewhere in this program. Thanks!

Reply:
Done